django_projects/locallibrary/catalog/admin.py

```diff
@@ -1,3 +1,31 @@
 from django.contrib import admin
+from .models import Author,Genre,Book,BookInstance
 
-# Register your models here.
+class AuthorAdmin(admin.ModelAdmin):
+    list_display = ('last_name', 'first_name', 'date_of_birth', 'date_of_death')
+    fields = ['first_name', 'last_name', ('date_of_birth', 'date_of_death')]
+
+admin.site.register(Author,AuthorAdmin)
+
+admin.site.register(Genre)
+
+class BookInstanceInline(admin.TabularInline):
+    model = BookInstance
+
+admin.site.register(Book)
+
+class BookAdmin(admin.ModelAdmin):
+    list_display = ('title', 'author', 'display_genre')
+    inlines = [BookInstanceInline]
+
+admin.site.register(BookInstance)
```

Comment:
chỗ này chưa đúng nhé, dùng decorator `@admin.register(model_name)` vì em có định nghĩa class `BookInstanceAdmin`

`admin.site.register(BookInstance)`: Đây là em đang khai báo theo cách thông thường rùi

Xem lại toàn bộ file này, dùng đồng nhất cách chị suggest ở trên nha